### PR TITLE
chore: CON-1341 Extract a tECDSA system test library and inline some tests

### PR DESCRIPTION
**Pull request by eichhorl**
In a future PR we will inline more tests and generalize the `ecdsa` naming to be tSchnorr inclusive

### CODE REVIEW

File: rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs [hunk 1]
```diff
@@ -1,13 +1,233 @@
...
+use registry_canister::mutations::do_add_nodes_to_subnet::AddNodesToSubnetPayload;
+use slog::{info, Logger};
+use std::collections::BTreeMap;
+
+const NODES_COUNT: usize = 4;
+const UNASSIGNED_NODES_COUNT: usize = 3;
+
+const ECDSA_KEY_TRANSCRIPT_CREATED: &str = "consensus_ecdsa_key_transcript_created";
+
+pub fn config(env: TestEnv) {
```
Comment (kpop-dfinity) [inline rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs:58]: nit: doesn't have to be `pub` anymore

File: rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs [hunk 2]
```diff
@@ -1,13 +1,233 @@
...
+            .for_each(|node| node.await_status_is_healthy().unwrap())
+    });
+    // Check all unassigned nodes are ready.
+    env.topology_snapshot().unassigned_nodes().for_each(|node| {
+        node.await_can_login_as_admin_via_ssh()
+            .expect("Timeout while waiting for all unassigned nodes to be ready.");
+    });
+}
+
+pub fn test(env: TestEnv) {
```
Comment (kpop-dfinity) [inline rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs:81]: ditto

also the same applies to the other inline tests

**Review by kpop-dfinity — APPROVED** [review]
Awesome, thanks :)

**Review by gregorydemay — APPROVED** [review]

File: rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs (hunk 2 above)
Comment (eichhorl) [reply rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs]: Thanks!

File: rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs [hunk 3]
```diff
@@ -1,13 +1,233 @@
...
+use registry_canister::mutations::do_add_nodes_to_subnet::AddNodesToSubnetPayload;
+use slog::{info, Logger};
+use std::collections::BTreeMap;
+
+const NODES_COUNT: usize = 4;
+const UNASSIGNED_NODES_COUNT: usize = 3;
+
+const ECDSA_KEY_TRANSCRIPT_CREATED: &str = "consensus_ecdsa_key_transcript_created";
+
+fn config(env: TestEnv) {
```
Comment (basvandijk) [inline rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs:58]: Consider renaming this from the old `config` naming convention to the modern:
```suggestion
fn setup(env: TestEnv) {
```

File: rs/tests/consensus/tecdsa/tecdsa_complaint_test.rs [hunk 4]
```diff
@@ -1,15 +1,94 @@
...
+    HasPublicApiUrl, HasTopologySnapshot, IcNodeContainer, NnsInstallationBuilder,
+};
 use ic_system_test_driver::systest;
-use ic_tests::tecdsa::tecdsa_complaint_test;
+use ic_system_test_driver::util::{assert_malicious_from_topo, runtime_from_url, MessageCanister};
+use ic_types::malicious_behaviour::MaliciousBehaviour;
+use ic_types::Height;
+use slog::info;
+
+fn config(env: TestEnv) {
```
Comment (basvandijk) [inline rs/tests/consensus/tecdsa/tecdsa_complaint_test.rs:22]:
```suggestion
fn setup(env: TestEnv) {
```

File: rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs [hunk 5]
```diff
@@ -1,13 +1,233 @@
...
+        }
+        tokio::time::sleep(std::time::Duration::from_secs(3)).await;
+    }
+}
 
 fn main() -> Result<()> {
     SystemTestGroup::new()
-        .with_setup(tecdsa::tecdsa_add_nodes_test::config)
-        .add_test(systest!(tecdsa::tecdsa_add_nodes_test::test))
+        .with_setup(config)
```
Comment (basvandijk) [inline rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs:229]:
```suggestion
        .with_setup(setup)
```

File: rs/tests/consensus/tecdsa/tecdsa_complaint_test.rs [hunk 6]
```diff
@@ -1,15 +1,94 @@
...
+    info!(logger, "Checking for malicious logs...");
+    assert_malicious_from_topo(&topology, vec!["maliciously_corrupt_idkg_dealings: true"]);
+    info!(logger, "Malicious log check succeeded.");
+}
 
 fn main() -> Result<()> {
     SystemTestGroup::new()
-        .with_setup(tecdsa_complaint_test::config)
-        .add_test(systest!(tecdsa_complaint_test::test))
+        .with_setup(config)
```
Comment (basvandijk) [inline rs/tests/consensus/tecdsa/tecdsa_complaint_test.rs:90]:
```suggestion
        .with_setup(setup)
```

File: rs/tests/consensus/tecdsa/tecdsa_remove_nodes_test.rs [hunk 7]
```diff
@@ -1,10 +1,142 @@
...
+use rand::seq::SliceRandom;
+use rand_chacha::ChaCha8Rng;
+use slog::info;
+
+const NODES_COUNT: usize = 4;
+const REMOVE_NODES_COUNT: usize = (NODES_COUNT / 3) + 1;
+// Seed for a random generator
+const RND_SEED: u64 = 42;
+
+fn config(env: TestEnv) {
```
Comment (basvandijk) [inline rs/tests/consensus/tecdsa/tecdsa_remove_nodes_test.rs]:
```suggestion
fn setup(env: TestEnv) {
```

File: rs/tests/consensus/tecdsa/tschnorr_message_sizes_test.rs [hunk 8]
```diff
@@ -1,18 +1,203 @@
...
+//         it to the management canister.
+const LOCAL_LIMIT: usize = 5 * MIB;
+const XNET_LIMIT: usize = 2 * MIB;
+
+fn make_schnorr_key_ids_for_all_algorithms() -> Vec<MasterPublicKeyId> {
+    vec![make_eddsa_key_id(), make_bip340_key_id()]
+}
+
+/// Creates one system subnet and two application subnets, the first one with schnorr keys enabled.
+fn config(env: TestEnv) {
```
Comment (basvandijk) [inline rs/tests/consensus/tecdsa/tschnorr_message_sizes_test.rs]:
```suggestion
fn setup(env: TestEnv) {
```

File: rs/tests/consensus/tecdsa/tschnorr_message_sizes_test.rs [hunk 9]
```diff
@@ -1,18 +1,203 @@
...
 fn main() -> Result<()> {
     SystemTestGroup::new()
-        .with_setup(tecdsa::tschnorr_message_sizes_test::config)
-        .add_test(systest!(
-            tecdsa::tschnorr_message_sizes_test::test_xnet_limit
-        ))
-        .add_test(systest!(
-            tecdsa::tschnorr_message_sizes_test::test_local_limit
-        ))
+        .with_setup(config)
```
Comment (basvandijk) [inline rs/tests/consensus/tecdsa/tschnorr_message_sizes_test.rs]:
```suggestion
        .with_setup(setup)
```

File: rs/tests/consensus/tecdsa/utils/BUILD.bazel [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    deps = [
+        # Keep sorted.
+        "//rs/canister_client",
+        "//rs/config",
+        "//rs/constants",
+        "//rs/nervous_system/common/test_keys",
+        "//rs/nns/cmc",
+        "//rs/nns/common",
+        "//rs/nns/constants",
+        "//rs/nns/governance",
```
Comment (basvandijk) [inline rs/tests/consensus/tecdsa/utils/BUILD.bazel:19]: Maybe consider being more explicit on [which rust_library](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/nns/governance/BUILD.bazel) you're depending here:
```suggestion
        "//rs/nns/governance:governance",
```

**Review by basvandijk — APPROVED** [review]
Nice!

File: rs/tests/consensus/tecdsa/utils/BUILD.bazel (hunk 10 above)
Comment (eichhorl) [reply rs/tests/consensus/tecdsa/utils/BUILD.bazel:19]: Seems like `buildifier` doesn't like that 😄 
```
==================== Test output for //bazel:buildifier_test:
./rs/tests/consensus/tecdsa/utils/BUILD.bazel:
--- ./rs/tests/consensus/tecdsa/utils/BUILD.bazel	2024-07-25 12:36:32.159122491 +0000
+++ /tmp/buildifier-tmp-3096240069	2024-07-25 12:39:47.775372690 +0000
@@ -16,7 +16,7 @@
         "//rs/nns/cmc",
         "//rs/nns/common",
         "//rs/nns/constants",
-        "//rs/nns/governance:governance",
+        "//rs/nns/governance",
         "//rs/nns/test_utils",
         "//rs/prep",
         "//rs/registry/canister",
exit status 1
```

**Pull request merged**